### PR TITLE
jsxy segmented control

### PR DESCRIPTION
**Pull request by jamesgorrie**
While trying to get whats-on rendering sorted, having this component JSXied would be very nice.

### CODE REVIEW

File: whats_on/app/views/components/whats-on-list-header/whats-on-list-header.njk [hunk 1]
```diff
@@ -35,7 +35,23 @@
...
+        {% componentJsx 'SegmentedControl', {
+          id: 'whatsOnFilter',
+          activeId: 'everything',
+          items: [{
+            id: 'everything',
+            url: '/whats-on',
+            text: 'Everything'
+          }, {
+            id: 'today',
+            url: '/whats-on/today',
```
Comment (jamesgorrie) [inline whats_on/app/views/components/whats-on-list-header/whats-on-list-header.njk]: TBD as per slack message in #domain-modelling.

File: common/views/components/SegmentedControl/SegmentedControl.js [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+          })}>See:</span>
+
+          <ul className='segmented-control__drawer-list js-segmented-control__drawer-list no-margin no-padding plain-list'>
+            {items.map((item, i) => (
+              <li
+                key={item.id}
+                className={classNames({
+                  [font({s: 'WB6'})]: true,
+                  [spacing({s: 3}, {padding: ['top', 'bottom']})]: true,
+                  'border-top-width-1': i === 0,
```
Comment (davidpmccormick) [inline common/views/components/SegmentedControl/SegmentedControl.js]: nice to see this in action

File: common/views/components/SegmentedControl/SegmentedControl.js [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+        'overflow-hidden': true,
+        'js-tablist segmented-control__list--inline': isTabControl
+      })}>
+        {items.map((item, i) => (
+          <li
+            key={item.id}
+            className={classNames({
+              [font({s: 'WB7'})]: true,
+              'border-right-width-1 border-right-color-black': i !== items.length - 1,
+              'segmented-control__item line-height-1 flex': true,
```
Comment (davidpmccormick) [inline common/views/components/SegmentedControl/SegmentedControl.js]: how do you decide whether each string should contain only one class?

Comment (jamesgorrie) [reply common/views/components/SegmentedControl/SegmentedControl.js]: I've been tending to favour 1 classname per property... thoughts?

Comment (davidpmccormick) [reply common/views/components/SegmentedControl/SegmentedControl.js]: Yeah, I'm down with that, but looks like a few (e.g. here) have made it into multiple classes per string

Comment (davidpmccormick) [reply common/views/components/SegmentedControl/SegmentedControl.js]: actually – I think it's only this line (the others make sense as they're instances of multiple classes being conditionally added based on some dynamic condition)

File: common/views/components/SegmentedControl/SegmentedControl.js [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+                category: 'component',
+                action: 'whats-on-daterange-picker:click',
+                label: 'title:' + item.text
+              })}
+              href={item.url}
+              className={classNames({
+                [spacing({s: 2}, {padding: ['top', 'right', 'bottom', 'left']})]: true,
+                'is-active bg-black font-white bg-hover-pewter': item.id === activeId,
+                'bg-white font-black bg-hover-pumice': item.id !== activeId,
+                'block plain-link': true,
```
Comment (davidpmccormick) [inline common/views/components/SegmentedControl/SegmentedControl.js]: …maybe this one too

**Pull request merged**